### PR TITLE
Dynamic support form

### PR DESCRIPTION
**Pull request by uzuntonev**
## Motivation and context

- Create dynamic SupportForm component with different steps
- Reach SupportForm example on route /support-form
- Entire form is translatable
- Usage for  collect information about each one who would like to support the association 
- Potential issues with form validation

## Example:
TPDkbr2GP9

### CODE REVIEW

File: src/components/support-form/SupportForm.tsx [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+  },
+  promoter: {
+    mediaPartner: false,
+    ambassador: false,
+    other: false,
+    otherText: '',
+  },
+}
+
+export default function Steppers(this: any) {
```
Comment (kachar) [inline src/components/support-form/SupportForm.tsx:213]: It's better to match the default export of the component with the filename `SupportForm`

This enhances the IDE suggestions and helps in refactorings

File: src/components/support-form/types/SuportFormData.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    phone: string
+    address: string
+  }
+  roles: {
+    benefactor: boolean
+    partner: boolean
+    volunteer: boolean
+    associationMember: boolean
+    promoter: boolean
+  }
```
Comment (kachar) [inline src/components/support-form/types/SuportFormData.ts:15]: Here's an idea about the types of roles

```suggestion
type RoleTypes = 'benefactor' | 'partner' | 'volunteer' | 'associationMember' | 'promoter'

export type SupportFormData = {
  terms: boolean
  info: {
    email: string
    name: string
    phone: string
    address: string
  }
  roles: { [key in RoleTypes]: boolean }
```

When we split a large type into smaller types it becomes easier to track the changes, reuse the types and see the whole picture

File: src/components/support-form/StepperIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  const classes = useStepIconStyles()
+  const { active, completed } = props
+
+  const icons: { [index: string]: React.ReactElement } = {
+    1: <SettingsIcon />,
+    2: <GroupAddIcon />,
+    3: <VideoLabelIcon />,
+    4: <SecurityIcon />,
+    5: <AccessibilityNew />,
+  }
```
Comment (kachar) [inline src/components/support-form/StepperIcon.tsx:42]: We've got a suggestion by @Svetlina89 to update the icons of the steps as follows:

```
1: `GroupAddIcon`
2: `HelpIcon`
3: `ContactMailIcon`
4: `SecurityIcon` (same)
5: `AccessibilityNew` (same)

**Review by kachar — APPROVED** [review]
Thanks for the great ambition, focus and execution around this core feature of the webapp :100: 

@uzuntonev @vikinatora The collaboration was strong with this one 

It looks and works quite nicely!
We'll be doing updates dictated by the product and UX teams on top of this effort quite freely.

I'm approving the PR, but feel free to push changes to the branch if you want to reflect on some points of the review. Will be merged a bit later on today.

File: src/components/support-form/SupportForm.tsx [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+      backgroundImage: 'linear-gradient( 95deg, #4AC3FF 0%, #29a2df 50%, #1b88be 100%)',
+    },
+  },
+  completed: {
+    '& $line': {
+      backgroundImage: 'linear-gradient( 95deg, #4AC3FF 0%, #29a2df 50%, #1b88be 100%)',
+    },
+  },
+  line: {
+    height: 3,
```
[the remark this answers is not in the document]
Comment (uzuntonev) [reply src/components/support-form/SupportForm.tsx:34]: The result of height: 3 should be 3px

**Pull request merged**